internal/core/data/rest_value_descriptor.go

```diff
@@ -0,0 +1,439 @@
+			http.Error(w, err.Error(), err.StatusCode)
+		default:
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+	}
+
+	pkg.Encode(vdList, w, LoggingClient)
+}
+
+// Return the value descriptors that are associated with the device specified by the device ID
+// Associated value descripts are expected parameters of PUT commands and expected results of PUT/GET commands
+// api/v1/valuedescriptor/deviceid/{id}
+func valueDescriptorByDeviceIdHandler(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+
+	vars := mux.Vars(r)
+
+	deviceId, err := url.QueryUnescape(vars["id"])
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		LoggingClient.Error("Error unescaping the device ID: " + err.Error())
+		return
+	}
+
+	ctx := r.Context()
+	// Get the value descriptors
+	vdList, err := getValueDescriptorsByDeviceId(deviceId, ctx)
+	if err != nil {
+		switch err := err.(type) {
+		case *types.ErrServiceClient:
+			http.Error(w, err.Error(), err.StatusCode)
+		case *errors.ErrDbNotFound:
+			http.Error(w, err.Error(), http.StatusNotFound)
+			return
+		default:
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+	}
+
+	pkg.Encode(vdList, w, LoggingClient)
+}
+
+// restValueDescriptorsUsageHandler checks if value descriptors are currently being used.
+// This functionality is useful for determining if a value descriptor can be updated, or deleted.
+// This functionality does not provide any guarantee that the value descriptor will not be in use in the near future.
+// Any functionality using the check to perform updates or deletes is responsible for handling any race conditions which
+// may occur.
+// Returns a map[string]bool where the key is the ValueDescriptor Name and the value is a bool stating if the
+// ValueDescriptor is currently in use.
+func restValueDescriptorsUsageHandler(w http.ResponseWriter, r *http.Request) {
+	qparams, err := url.ParseQuery(r.URL.RawQuery)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	namesFilter := qparams[NAMES]
+	var vds []contract.ValueDescriptor
+	var op value_descriptor.GetValueDescriptorsExecutor
+	if len(namesFilter) <= 0 {
+		// We are not filtering so get all the value descriptors
+		op = value_descriptor.NewGetValueDescriptorsExecutor(dbClient, LoggingClient, Configuration.Service)
+	} else {
+		op = value_descriptor.NewGetValueDescriptorsNameExecutor(strings.Split(namesFilter[0], ","), dbClient, LoggingClient, Configuration.Service)
+	}
+
+	vds, err = op.Execute()
+	if err != nil {
+		switch err.(type) {
+		case *errors.ErrLimitExceeded:
+			http.Error(w, err.Error(), http.StatusRequestEntityTooLarge)
+		default:
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+
+		return
+	}
+
+	// Use this data structure so that we can obtain the desired JSON format. Please see RAML for response format
+	// information.
+	resp := make([]map[string]bool, 0)
```

Comment:
I had to do this in order to get the desired response format. I am not 100% a fan of this. If someone can suggest something else, I'd like to readdress this.

Reply:
`var resp []map[string]bool`

Reply:
@AnthonyMBonafide You know what? The previous way you had this defined was better because now if there are no results a literal "null" is returned in the response instead of the empty map.

Reply:
Good catch! I missed that.